modules/experimental-layers/src/tile-layer/tile-layer.md

```diff
@@ -0,0 +1,236 @@
+# TileLayer
+
+This TileLayer takes in a function `getTileData` that fetches tiles, and renders it in a GeoJsonLayer or with the layer returned in `renderSubLayers`.
+
+```js
+import DeckGL from 'deck.gl';
+import TileLayer from '@deck.gl/experimental-layers/tile-layer/tile-layer';
+
+getTileData({x, y, z}) {
+    const mapSource = `https://a.tiles.mapbox.com/v4/mapbox.mapbox-streets-v7/${z}/${x}/${y}.vector.pbf?access_token=${MAPBOX_TOKEN}`;
+    return fetch(mapSource)
+      .then(response => {
+        return response.arrayBuffer();
+      })
+      .then(buffer => {
+        return decodeTiles(x, y, z, buffer);
+      });
+  }
+
+  render() {
+    return (
+      <DeckGL
+        layers={[
+          new TileLayer({
+            ...MAP_LAYER_STYLES,
```

Comment:
This constant is undefined in this code sample -- just expand it in place with a few examples.

Reply:
In this code sample, is it necessary to process the fetched tile data? (`decodeTile` is not defined in this snippet either)

Reply:
Let's add a minimal version of the decoder:

```js
import {VectorTile} from '@mapbox/vector-tile';
import Protobuf from 'pbf';

function vectorTileToGeoJSON(buffer, x, y, z) {
  const tile = new VectorTile(new Protobuf(buffer));
  const features = [];

  for (const layerName in tile.layers) {
    const vectorTileLayer = tile.layers[layerName];

    for (let i = 0; i < vectorTileLayer.length; i++) {
      const vectorTileFeature = vectorTileLayer.feature(i);
      const feature = vectorTileFeature.toGeoJSON(x, y, z);
      features.push(feature);
    }
  }

  return features;
}
```